### PR TITLE
Add CSISnapshotTimeout to the backup in tests

### PR DESCRIPTION
**Pull request by mpryc**
This adds 3min timeout to the CSISnapshotTimeout within tests.

### CODE REVIEW

**Comment by weshayutin** [comment]
thank you!

File: tests/e2e/lib/backup.go [hunk 1]
```diff
@@ -12,7 +16,55 @@ import (
...
+	case "m":
+		dur = time.Duration(value) * time.Minute
+	case "h":
+		dur = time.Duration(value) * time.Hour
+	default:
+		return metav1.Duration{}, errors.New("Invalid duration suffix: " + suffix)
+	}
+
+	return metav1.Duration{Duration: dur}, nil
+}
```
Comment (kaovilai) [inline tests/e2e/lib/backup.go:55]: If you're importing time package, just use it's parse function no?
```suggestion
// ParseDuration parses a duration string.
// A duration string is a possibly signed sequence of
// decimal numbers, each with optional fraction and a unit suffix,
// such as "300ms", "-1.5h" or "2h45m".
// Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
func parseDuration(durationStr string) (metav1.Duration, error) {
	time, err := time.ParseDuration(durationStr)
	return metav1.Duration{Duration: time}, err
}
```

File: tests/e2e/lib/backup.go [hunk 2]
```diff
@@ -12,7 +16,55 @@ import (
...
+
+func CreateBackupForNamespaces(ocClient client.Client, veleroNamespace, backupName string, namespaces []string, defaultVolumesToFsBackup bool, snapshotMoveData bool, csiSnapshotTimeout string) (velero.Backup, error) {
+
+	csiSnapDuration, err := parseDuration(csiSnapshotTimeout)
+
+	if err != nil {
+		log.Printf("Error parsing duration: %v", err)
+		return velero.Backup{}, err
+	}
```
Comment (kaovilai) [inline tests/e2e/lib/backup.go:64]: Or just do this.
```suggestion
	time, err := time.ParseDuration(durationStr)
	if err != nil {
		log.Printf("Error parsing duration: %v", err)
		return velero.Backup{}, err
	}
	csiSnapDuration := metav1.Duration{Duration: time}
```

File: tests/e2e/lib/backup.go [hunk 3]
```diff
@@ -12,7 +14,41 @@ import (
...
+// - <value> is a positive integer representing the duration value.
+// - <suffix> is a character representing the duration unit:
+//   - "s" for seconds
+//   - "m" for minutes
+//   - "h" for hours
+//
+// Example valid inputs: "10s", "5m", "1h"
+//
+// Returns the parsed duration as a v1.Duration object, or an error if parsing fails.
+func parseDuration(durationStr string) (metav1.Duration, error) {
```
Comment (kaovilai) [inline tests/e2e/lib/backup.go:30]:
```suggestion
// parseDuration parses a duration string and return k8s.io/apimachinery/pkg/apis/meta/v1.Duration type.
// A duration string is a possibly signed sequence of
// decimal numbers, each with optional fraction and a unit suffix,
// such as "300ms", "-1.5h" or "2h45m".
// Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
//
// Example valid inputs: "10s", "5m", "1h"
//
// Returns the parsed duration as a v1.Duration object, or an error if parsing fails.
func parseDuration(durationStr string) (metav1.Duration, error) {
```

**Review by kaovilai — APPROVED** [review]

**Comment by kaovilai** [comment]
```
      Operation Error:  Failed to check and update snapshot content: failed to remove VolumeSnapshotBeingCreated annotation on the content snapcontent-792ddcf6-5b8c-4e7c-a65b-adf0d5a76e17: "snapshot controller failed to update snapcontent-792ddcf6-5b8c-4e7c-a65b-adf0d5a76e17 on API server: Operation cannot be fulfilled on volumesnapshotcontents.snapshot.storage.k8s.io \"snapcontent-792ddcf6-5b8c-4e7c-a65b-adf0d5a76e17\": the object has been modified; please apply your changes to the latest version and try again"
```
still happening.

**Comment by kaovilai** [comment]
I'm not sure the flake retrying is working to resolve the csi specific issues.. there must be something else we can do.

**Comment by kaovilai** [comment]
Same issue that is being cherry-picked here https://github.com/openshift/csi-external-snapshotter/pull/140
```
      Operation Error:  Failed to check and update snapshot content: failed to remove VolumeSnapshotBeingCreated annotation on the content snapcontent-792ddcf6-5b8c-4e7c-a65b-adf0d5a76e17: "snapshot controller failed to update snapcontent-792ddcf6-5b8c-4e7c-a65b-adf0d5a76e17 on API server: Operation cannot be fulfilled on volumesnapshotcontents.snapshot.storage.k8s.io \"snapcontent-792ddcf6-5b8c-4e7c-a65b-adf0d5a76e17\": the object has been modified; please apply your changes to the latest version and try again"
```
https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_oadp-operator/1317/pull-ci-openshift-oadp-operator-master-4.12-e2e-test-aws/1755263202590986240#1:build-log.txt%3A483

**Comment by kaovilai** [comment]
>As you can see we already cover this in our flake detection.
I know it's covered.. but just wanna point it out so that cp PR has more references. In addition, I have seen 3 retries not resolve the issue.

File: tests/e2e/backup_restore_suite_test.go [hunk 4]
```diff
@@ -148,11 +148,11 @@ func runBackupAndRestore(brCase BackupRestoreCase, expectedErr error, updateLast
 	time.Sleep(brCase.AppReadyDelay)
 	// create backup
 	log.Printf("Creating backup %s for case %s", backupName, brCase.Name)
-	backup, err := CreateBackupForNamespaces(dpaCR.Client, namespace, backupName, []string{brCase.ApplicationNamespace}, brCase.BackupRestoreType == RESTIC || brCase.BackupRestoreType == KOPIA, brCase.BackupRestoreType == CSIDataMover)
+	backup, err := CreateBackupForNamespaces(dpaCR.Client, namespace, backupName, []string{brCase.ApplicationNamespace}, brCase.BackupRestoreType == RESTIC || brCase.BackupRestoreType == KOPIA, brCase.BackupRestoreType == CSIDataMover, "180s")
```
Comment (mateusoliveira43) [inline tests/e2e/backup_restore_suite_test.go:151]: we use `time.<unit>` in a lot of places, why use a string here?

Comment (mpryc) [reply tests/e2e/backup_restore_suite_test.go:151]: To easy write tests and align with what server flags are expected in upstream velero. Also align with the velero crd which is string:

https://github.com/vmware-tanzu/velero/pull/5104/files#diff-1d9e81ef7720633f549a8239f569a20ebd656c9d880be086d44e18954a035cc8R44

Comment (mateusoliveira43) [reply tests/e2e/backup_restore_suite_test.go:151]: change to `3*time.Minute`, delete parse function and use `String()` method from `time` package

**Review by mateusoliveira43 — CHANGES_REQUESTED** [review]
simplify code using time tools

**Pull request closed without merging**